src/clj/rems/api/permissions.clj

```diff
@@ -25,13 +28,22 @@
      :tags ["permissions"]
      (GET "/:user" []
        ;; We're trying to replicate https://github.com/CSCfi/elixir-rems-proxy/#get-permissionsusername here
-       :summary "Experimental. Returns user's permissions in ga4gh visa format. See also https://github.com/CSCfi/rems/blob/master/docs/ga4gh-visas.md"
-       :roles #{:handler :owner}
+       :summary (str "Experimental. Returns user's permissions in ga4gh visa format. "
+                     "Handlers, owners and reporters can query anybody's permissions. Other users can query their own permissions. "
+                     "See also https://github.com/CSCfi/rems/blob/master/docs/ga4gh-visas.md")
+       :roles #{:logged-in}
        :path-params [user :- (describe s/Str "return permissions for this user, required")]
        :query-params [{expired :- (describe s/Bool "whether to include expired permissions") false}]
        :return GetPermissionsResponse
        (if (not (:enable-permissions-api config/env))
          (not-implemented "permissions api not implemented")
-         (if (empty? user)
+         (cond
+           (empty? user)
            (api-util/not-found-json-response)
+
+           (not (or (has-roles?  :handler :owner :organization-owner :reporter)
+                    (= user (getx-user-id))))
+           (throw-forbidden)
+
+           :else
```

Comment:
It's slightly unfortunate that some of these return and some throw. This expression is not quite a clean `or` or `cond`. The `not` `or` part is also rather difficult to read.

Can't see an immediate easy refactoring though.

Reply:
I found it more natural to write the permission checking as `(not (allowed))`. I'll see if I can change the throw into a response.

Reply:
We do the logging of the forbiddens in rems.api/forbidden-handler, so if I produce a forbidden response here it'll be slightly different from throwing an exception. Also, we just use throw-forbidden everywhere. So no fix for now.